### DISCUSSION TITLE
Unable to set user defined environment variable in KC

### DISCUSSION THREAD

**Discussion by williamyeungforsure** [strimzi/strimzi-kafka-operator#7134] [Q&A]
Hi there,
I am trying to set the namespace as an environment variable in KafkaConnect, so that I could use it to format a string that would be used in the KafkaConnector's `spec.config` like this: "postgresql-ha-${NAMESPACE}"
At first, I defined it inside KafkaConnect's `externalConfiguration.env` using `Downward API` like this:
```
- name: NAMESPACE
   valueFrom:
       fieldRef:
         fieldPath: metadata.namespace
```
However, seems like Strimzi API does not support this fieldRef syntax. I got such error here:
```
error validating data: ValidationError(KafkaConnect.spec.externalConfiguration.env[0].valueFrom): unknown field "fieldRef" in io.strimzi.kafka.v1beta2.KafkaConnect.spec.externalConfiguration.env.valueFrom
```
Does anyone know there's a way to set a user defined environment variable in KafkaConnect? Or is there a way to get the namespace of the deployed KafkaConnector pod? Thanks a lot.

**Comment by scholzj** [comment]
You can set environment variables either by setting the value directly or based on Secrets or ConfigMaps. You cannot set them based on the Downward API.
